commands/build.go

```diff
@@ -701,9 +702,19 @@ func parseInvokeConfig(invoke string) (cfg invokeConfig, err error) {
 		value := parts[1]
 		switch key {
 		case "args":
-			cfg.Cmd = append(cfg.Cmd, value) // TODO: support JSON
+			var alist []string
+			if err := json.Unmarshal([]byte(value), &alist); err == nil {
+				cfg.Cmd = append(cfg.Cmd, alist...)
+			} else {
+				cfg.Cmd = append(cfg.Cmd, value)
+			}
 		case "entrypoint":
-			cfg.Entrypoint = append(cfg.Entrypoint, value) // TODO: support JSON
+			var elist []string
+			if err := json.Unmarshal([]byte(value), &elist); err == nil {
+				cfg.Entrypoint = append(cfg.Entrypoint, elist...)
+			} else {
+				cfg.Entrypoint = append(cfg.Entrypoint, value)
+			}
 		case "env":
 			cfg.Env = append(cfg.Env, value)
```

Comment:
IMO, this should also be supported for `env` as well.

Reply:
Thanks for the comment. Added JSON array support to env as well.

```
$ BUILDX_EXPERIMENTAL=1 /tmp/out/buildx build --invoke 'entrypoint=["sh"],"args=[""-c"", ""env | grep -e FOO -e AAA""]","env=[""FOO=bar"", ""AAA=bbb""]"' /tmp/ctx3
[+] Building 2.1s (9/9) FINISHED                       docker-container:latest
 => [internal] connecting to local controller                             0.0s
 => [internal] load build definition from Dockerfile                      0.1s
 => => transferring dockerfile: 100B                                      0.0s
 => [internal] load metadata for docker.io/library/busybox:latest         1.5s
 => [auth] library/busybox:pull token for registry-1.docker.io            0.0s
 => [internal] load .dockerignore                                         0.1s
 => => transferring context: 2B                                           0.0s
 => [1/4] FROM docker.io/library/busybox@sha256:560af6915bfc8d7630e50e21  0.0s
 => => resolve docker.io/library/busybox@sha256:560af6915bfc8d7630e50e21  0.0s
 => CACHED [4/4] RUN echo 3 > /3                                          0.0s
 => CACHED [2/4] RUN echo hi > /hi                                        0.0s
 => CACHED [3/4] RUN echo 2 > /2                                          0.0s
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
Launching interactive container. Press Ctrl-a-c to switch to monitor console
Interactive container was restarted with process "hwg57oeol5m2rozqz6iw2f3yj". Press Ctrl-a-c to switch to the new container
FOO=bar
AAA=bbb

```